jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/ServletPathSpec.java

```diff
@@ -270,6 +284,44 @@ private boolean isWildcardMatch(String path)
         return false;
     }
 
+    @Override
+    public MatchedPath matched(String path)
+    {
+        switch (_group)
+        {
+            case EXACT:
+                if (_declaration.equals(path))
+                    return new ServletMatchedPath(path, null); // TODO: return final matchedpath
+                break;
+            case PREFIX_GLOB:
+                if (isWildcardMatch(path))
+                {
+                    String pathMatch = path;
+                    String pathInfo = null;
+                    if (path.length() != (_specLength - 2))
+                    {
+                        pathMatch = path.substring(0, _specLength - 2);
+                        pathInfo = path.substring(_specLength - 2);
+                    }
+                    return new ServletMatchedPath(pathMatch, pathInfo);
+                }
+                break;
+            case SUFFIX_GLOB:
+                if (path.regionMatches((path.length() - _specLength) + 1, _declaration, 1, _specLength - 1))
+                    return new ServletMatchedPath(path, null);
+                break;
+            case ROOT:
+                // Only "/" matches
+                if ("/".equals(path))
+                    return new ServletMatchedPath("", path); // TODO: review this
```

Comment:
bizarrely I think that is correct!  "" matches "/" and returns "" as the match and "/" as the part after the match !?!?!?!?

Reply:
Now using `MatchedPath.from("", path);`

Reply:
The logic here is identical to how it was in 9.4.46 too.

See: 

* https://github.com/eclipse/jetty.project/blob/jetty-9.4.46.v20220331/jetty-http/src/test/java/org/eclipse/jetty/http/pathmap/ServletPathSpecTest.java#L122
* https://github.com/eclipse/jetty.project/blob/jetty-9.4.46.v20220331/jetty-http/src/test/java/org/eclipse/jetty/http/pathmap/ServletPathSpecTest.java#L152-L153

https://github.com/eclipse/jetty.project/blob/bc17a0369a11ecf40bb92c839b9ef0a8ac50ea18/jetty-http/src/test/java/org/eclipse/jetty/http/pathmap/ServletPathSpecTest.java#L122

https://github.com/eclipse/jetty.project/blob/bc17a0369a11ecf40bb92c839b9ef0a8ac50ea18/jetty-http/src/test/java/org/eclipse/jetty/http/pathmap/ServletPathSpecTest.java#L152-L153

It surprised me too, and we have many tests elsewhere in jetty that test for ROOT path hits.